File: execution/env_helper.js
/**
 * Node.js helper utility for loading environment variables and configuration for Layer 3 execution scripts.
 */

import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

export const BASE_DIR = path.resolve(__dirname, '..');
export const TMP_DIR = path.resolve(BASE_DIR, '.tmp');
export const ENV_FILE = path.resolve(BASE_DIR, '.env');

export function ensureTmpDir() {
  if (!fs.existsSync(TMP_DIR)) {
    fs.mkdirSync(TMP_DIR, { recursive: true });
  }
  return TMP_DIR;
}

export function loadEnv(envPath = ENV_FILE) {
  const envVars = {};
  if (!fs.existsSync(envPath)) {
    return envVars;
  }

  const content = fs.readFileSync(envPath, 'utf-8');
  for (const line of content.split('\n')) {
    const trimmed = line.trim();
    if (!trimmed || trimmed.startsWith('#') || !trimmed.includes('=')) {
      continue;
    }
    const [key, ...vals] = trimmed.split('=');
    const val = vals.join('=').trim().replace(/^['"]|['"]$/g, '');
    const cleanKey = key.trim();
    envVars[cleanKey] = val;
    if (!process.env[cleanKey]) {
      process.env[cleanKey] = val;
    }
  }
  return envVars;
}

export function getEnvVar(key, defaultValue = null) {
  loadEnv();
  return process.env[key] || defaultValue;
}

if (process.argv[1] === fileURLToPath(import.meta.url)) {
  ensureTmpDir();
  const loaded = loadEnv();
  console.log(`[OK] Layer 3 environment helper (Node.js) initialized. Loaded ${Object.keys(loaded).length} keys from ${path.basename(ENV_FILE)}.`);
}

File: execution/env_helper.py
"""
Helper utility for loading environment variables and configuration for Layer 3 execution scripts.
"""

import os
from pathlib import Path

BASE_DIR = Path(__file__).resolve().parent.parent
TMP_DIR = BASE_DIR / ".tmp"
ENV_FILE = BASE_DIR / ".env"

def ensure_tmp_dir() -> Path:
    """Ensure .tmp directory exists and return its path."""
    TMP_DIR.mkdir(parents=True, exist_ok=True)
    return TMP_DIR

def load_env(env_path: Path = ENV_FILE) -> dict:
    """
    Simple, zero-dependency .env parser that loads key-value pairs
    into os.environ if not already set.
    """
    env_vars = {}
    if not env_path.exists():
        return env_vars

    with open(env_path, "r", encoding="utf-8") as f:
        for line in f:
            line = line.strip()
            if not line or line.startswith("#") or "=" not in line:
                continue
            key, val = line.split("=", 1)
            key = key.strip()
            val = val.strip().strip("'\"")
            env_vars[key] = val
            if key not in os.environ:
                os.environ[key] = val
    return env_vars

def get_env_var(key: str, default: str = None) -> str:
    """Retrieve environment variable with fallback default."""
    load_env()
    return os.environ.get(key, default)

if __name__ == "__main__":
    ensure_tmp_dir()
    loaded = load_env()
    print(f"[OK] Layer 3 environment helper initialized. Loaded {len(loaded)} keys from {ENV_FILE.name}.")
